src/NuGet.Core/NuGet.Configuration/Settings/Items/SourceItem.cs

```diff
@@ -77,8 +77,8 @@ public override bool Equals(object other)
                 return true;
             }
 
-            return string.Equals(Key, source.Key, StringComparison.Ordinal) &&
```

Comment:
I decided to revert this change.  For below input:
```
  <packageSources>
    <clear />
    <add key="encyclopaedia" value="C:\Users\eryondon\Desktop\NamespaceTestPackages\PublicRepository" />
    <add key="Encyclopaedia" value="https://api.nuget.org/v3/index.json" />
  </packageSources>
```
Before this change:
![image](https://user-images.githubusercontent.com/8766776/133527007-6eb41e5a-0786-4179-8a0f-e3e48f7ef269.png)

After this change:
![image](https://user-images.githubusercontent.com/8766776/133526717-39fbc72c-efe0-4b53-ac83-ac5db890a54f.png)

Please note source url is quite different and looks breaking change.